ee/clickhouse/models/property.py

```diff
@@ -103,13 +105,28 @@ def parse_prop_clauses(
             final.append(f"{filter_query} AND {table_name}team_id = %(team_id)s" if team_id else filter_query)
             params.update(filter_params)
         elif prop.type == "group":
-            # :TRICKY: This assumes group properties have already been joined, as in trends query
-            filter_query, filter_params = prop_filter_json_extract(
-                prop, idx, prepend, prop_var=f"group_properties_{prop.group_type_index}", allow_denormalized_props=False
-            )
-
-            final.append(filter_query)
-            params.update(filter_params)
+            if group_properties_joined:
+                filter_query, filter_params = prop_filter_json_extract(
+                    prop,
+                    idx,
+                    prepend,
+                    prop_var=f"group_properties_{prop.group_type_index}",
+                    allow_denormalized_props=False,
+                )
+                final.append(filter_query)
+                params.update(filter_params)
+            else:
+                # :TRICKY: offer groups support for queries which don't support it yet (e.g. lifecycle)
```

Comment:
Very confusing.

I think this is what it means?

```suggestion
                # :TRICKY: offer groups global filtering support for queries which don't support groups querying yet (e.g. lifecycle)
```